app/src/main/java/org/dhis2/uicomponents/map/managers/EventMapManager.kt

```diff
@@ -65,6 +107,31 @@ class EventMapManager(mapView: MapView) : MapManager(mapView) {
         }
     }
 
+    override fun findFeatures(
+        source: String,
+        propertyName: String,
+        propertyValue: String
+    ): List<Feature>? {
+        return mutableListOf<Feature>().apply {
+            featureCollection?.features()?.filter {
+                it.getStringProperty(propertyName) == propertyValue
+            }?.map {
+                mapLayerManager.getLayer(LayerType.EVENT_LAYER.name)
+                    ?.setSelectedItem(it)
+                it
+            }?.let { addAll(it) }
+            deFeatureCollection.values.map { collection ->
```

Comment:
no need to change, but you can use mapValues directly instead of values.map :+1:

Reply:
mapValues returns the entry (key and value) not just the value